### DISCUSSION TITLE
Sticky content tabs

### DISCUSSION THREAD

**Discussion by sriumcp** [squidfunk/mkdocs-material#2676] [Ideas]
Thanks for `mkdocs-material`  which makes it super easy for developers to impress and delight end-users with clean documentation.

***

I had a question on content tabbing.

Is there a way to create "sticky content tabs" as follows? 

When I choose "C" in hello world section on top, I also want "C" to be chosen in the good bye world section at the bottom. Ditto for C++.

***

## Hello world
=== "C"

    ``` c
    #include <stdio.h>

    int main(void) {
      printf("Hello world!\n");
      return 0;
    }
    ```

=== "C++"

    ``` c++
    #include <iostream>

    int main(void) {
      std::cout << "Hello world!" << std::endl;
      return 0;
    }
    ```

## Intervening material
A bunch of text here... because ... there are other things that intervene... and need to be said.

## Goodbye world
=== "C"

    ``` c
    #include <stdio.h>

    int main(void) {
      printf("Goodbye world!\n");
      return 0;
    }
    ```

=== "C++"

    ``` c++
    #include <iostream>

    int main(void) {
      std::cout << "Goodbye world!" << std::endl;
      return 0;
    }
    ```

**Comment by facelessuser** [comment]
You could probably do this with Javascript and attach some events to the radio boxes that are used to change the tabs. Every time a tab is changed, you'd just have the JavaScript update the others on the page. Theoretically, you could also save that state in localstorage and retrieve that info on page load and carry that state across all pages.

Whether Material itself would want to ship such functionality in the theme is another question.

**Reply by sriumcp** [reply]
Thank you @facelessuser.

I would love to explore the javascript option here (along with the events attached to the radio boxes).

Is there a snippet that you could share that could perhaps put me on the right track?

Even if material doesn't ship this functionality out of the box, perhaps it would be good to document this somewhere once I figure this out/get it working (documentation can simply live in this thread to begin with).

**Reply by facelessuser** [reply]
Unfortunately, I have not actually implemented such functionality, so I do not have a snippet handy. I am confident this could be implemented, but I haven't had the need to actually code up such functionality.

**Reply by squidfunk** [reply]
I wanted to build exactly this feature for some time, just haven’t had the chance.

Could I ask you to create an issue with a feature request?

**Reply by sriumcp** [reply]
With pleasure!

**Reply by sriumcp** [reply]
Done. https://github.com/squidfunk/mkdocs-material/issues/2678

**Comment by squidfunk** [comment]
[Linked content tabs](https://squidfunk.github.io/mkdocs-material/reference/content-tabs/#linking-content-tabs) are available as of Insiders 2.9.0!
